pkg/apis/core/validation/validation.go

```diff
@@ -2312,11 +2311,14 @@ func ValidatePersistentVolumeClaimStatusUpdate(newPvc, oldPvc *core.PersistentVo
 		allErrs = append(allErrs, validateBasicResource(qty, capPath.Key(string(r)))...)
 	}
 	if validationOpts.EnableRecoverFromExpansionFailure {
-		resizeStatusPath := field.NewPath("status", "resizeStatus")
-		if newPvc.Status.ResizeStatus != nil {
-			resizeStatus := *newPvc.Status.ResizeStatus
-			if !resizeStatusSet.Has(string(resizeStatus)) {
-				allErrs = append(allErrs, field.NotSupported(resizeStatusPath, resizeStatus, resizeStatusSet.List()))
+		resizeStatusPath := field.NewPath("status", "allocatedResourceStatus")
+		if newPvc.Status.AllocatedResourceStatus != nil {
+			resizeStatus := newPvc.Status.AllocatedResourceStatus
+			for k, v := range resizeStatus {
+				if !resizeStatusSet.Has(string(v)) {
```

Comment:
Note to reviewers: I am not restricting the keys but I am restricting the values. So basically I do not expect an resource associated with pvc to be in arbitrary resizing state. It can only be in one of the predefined states.

Reply:
I kept the keys unvalidated because of https://github.com/kubernetes/enhancements/pull/3780/files#diff-0ac6b83209829fbf96fafea2c79705f00e9b0fee254d4f9b06abffaf2cd5effdR108 comment here. @mattcary wanted vendors to add arbitrary resources that needs to be allocated for PVC and those can be resized.

if we are going to validate keys, we will be closing that door. I am okay with it for now, we can always revisit and relax that. @mattcary what do you think?

Reply:
I'm proposing that for keys that are unprefixed or use k8s.io prefix, those need to be well-known. If someone wants to add their own custom keys, then they should use their own prefix.

Reply:
Added

Reply:
It can only be in one of the predefined states <-- if there are two different resources being updated in one operation, should the state be just one of the resource in progress and proceed with the update?